src/linter/lintWorkspace.ts

```diff
@@ -0,0 +1,29 @@
+import {AbstractAdapter} from "@ui5/fs";
+import lintXml from "./xmlTemplate/linter.js";
+import lintJson from "./manifestJson/linter.js";
+import lintHtml from "./html/linter.js";
+import {taskStart} from "../util/perf.js";
+import TypeLinter from "./ui5Types/TypeLinter.js";
+import LinterContext, {LintResult, LinterParameters, LinterOptions} from "./LinterContext.js";
+
+export default async function lintWorkspace(
+	workspace: AbstractAdapter, options: LinterOptions
+): Promise<LintResult[]> {
+	const done = taskStart("Linting Workspace");
+
+	const context = new LinterContext(options);
+	const params: LinterParameters = {
+		workspace, context,
+	};
+
+	await Promise.all([
+		lintXml(params),
+		lintJson(params),
+		lintHtml(params),
+	]);
```

Comment:
With the former implementation those were returning the transpiled source code, messages and source maps.

We might need those in the future as we would like to eventually analyze html's script content, build & check code against manifest.json directives etc.

I'm not sure I understand how this is addressed in that PR.
Currently the TypeLinter checks for `js/ts` files and analyzes them. However, I'm missing the mechanism that would bring the transpiled `html` resources, for example to that analysis there.

Reply:
Here are some examples that this functionliaty is missing with this PR:

https://github.com/SAP/ui5-linter/pull/69/files#diff-2962e8dc8da712ee775c89026ca06b8676239e83618b45f89e98959640009731L68

https://github.com/SAP/ui5-linter/pull/69/files#diff-8fa33d71c5c9df86053ff360b697c8987242d7a12895fb0d8e5a8dad850d0751L1163

Reply:
As an example, have a look at the XML-Linter in this PR:
https://github.com/SAP/ui5-linter/pull/69/files#diff-1bc919fd57fdbc38b20c5feaca329ba1764224846e55d4a8b7b654f18e9ca35cR29-R40

The transpiled resources are written to the `workspace` from where the TypeLinter will later read them together will all other ".js" resources.

The association back to the original (XML)-Resource is done by reading the original  file name from the "sources" array of the associated source map in the TypeLinter's reporter: https://github.com/SAP/ui5-linter/pull/69/files#diff-1a8fcfb41f5f0e0d7d87295b6290a8af77f0d01a7881196967276fa790410ebbR148

I think the same can be applied for the HTML linter:
1. src/linter/html/transpiler.ts returns one or more JavaScript + SourceMap pairs
2. src/linter/html/linter.ts creates `@ui5/fs/Resource` instances and writes them to the workspace
3. TypeLinter picks them up with all the other JS files